cmd/peggo/ethrpc_manager.go

```diff
@@ -0,0 +1,138 @@
+package peggo
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/knadh/koanf"
+	"github.com/pkg/errors"
+)
+
+type EthRPCManager struct {
+	currentEndpoint int // index (in the slice of configured RPC endpoints) of most recent endpoint used
+	client          *rpc.Client
+	konfig          *koanf.Koanf
+}
+
+// creates an instance of EthRPCManager with a given konfig.
+func NewEthRPCManager(konfig *koanf.Koanf) *EthRPCManager {
+	ethManager := &EthRPCManager{
+		konfig: konfig,
+	}
+	return ethManager
+}
+
+// closes and sets to nil the stored eth RPC client
+func (em *EthRPCManager) CloseClient() {
+	if em.client != nil {
+		em.client.Close()
+		em.client = nil
+	}
+}
+
+// closes the current client and dials configured ethereum rpc endpoints in a roundrobin fashion until one
+// is connected. returns an error if no endpoints ar configured or all dials failed
+func (em *EthRPCManager) DialNext() error {
+	if em.konfig == nil {
+		return errors.New("ethRPCManager konfig is nil")
+	}
+
+	rpcs := strings.Split(strings.ReplaceAll(em.konfig.String(flagEthRPCs), " ", ""), ",")
+
+	em.CloseClient()
+
+	dialIndex := func(i int) bool {
+		if cli, err := rpc.Dial(rpcs[i]); err == nil {
+			em.currentEndpoint = i
+			em.client = cli
+			return true
+		}
+		fmt.Fprintf(os.Stderr, "Failed to dial to Ethereum RPC: %s\n", rpcs[i])
```

Comment:
We should probably pass the core logger to `NewEthRPCManager` instead of using `fmt.*`.